### PR TITLE
Update README to reference Omax.cfg and OmaxLTO.cfg

### PR DESCRIPTION
**Pull request by nasherm**
Updating the README.md file to detail use case of Omax.cfg and OmaxLTO.cfg.

### CODE REVIEW

**Review by stuij — COMMENTED** [review]
I feel we should warn the reader that the optimisations include `-ffast-math`, which breaks IEEE compliance, amongst others.

File: README.md [hunk 1]
```diff
@@ -140,6 +140,19 @@ $ clang \
 -o example example.c
 ```
 
+The builds of the toolchain come packaged with two config files, Omax.cfg and OmaxLTO.cfg.
+When used, these config files enable several build optimisation flags. OmaxLTO.cfg enables LTO
```
Comment (voltur01) [inline README.md:144]: Would be nice to explain:
...optimization flags to achieve highest performance on typical embedded benchmarks.

Comment (voltur01) [inline README.md:144]: Any recommendation when to use them (LTO and no LTO) together or separately?

Comment (voltur01) [inline README.md:144]: Would be also nice to mention that the flags used are internal ones, any advice where to look to explanation of what these options are doing?

File: README.md [hunk 2]
```diff
@@ -140,6 +140,19 @@ $ clang \
...
+
+```
+$ clang \
+example.c \
+...
+--config=Omax.cfg \
+--config=OmaxLTO.cfg \
+-o example
+```
+
```
Comment (voltur01) [inline README.md:155]: Could we also add a hint like it is possible to use these config files as an inspiration to create config file for your project by picking options from there and tuning the parameters as fits the project?

**Review by voltur01 — COMMENTED** [review]
Thank you for adding the details! Please see some ideas how to expand it a bit.

File: README.md (hunk 1 above)
Comment (nasherm) [reply README.md]: Done!

Comment (nasherm) [reply README.md]: I've updated and provided a link to clang documentation in my newest patch.

Comment (nasherm) [reply README.md]: I've made reference to how LTO can potentially increase link time and memory usage during linking

File: README.md (hunk 2 above)
Comment (nasherm) [reply README.md:159]: Done!

**Comment by nasherm** [comment]
> I feel we should warn the reader that the optimisations include `-ffast-math`, which breaks IEEE compliance, amongst othe

@stuij Provided a brief explanation of the risk of `-ffast-math` in my newest patch

File: README.md [hunk 3]
```diff
@@ -143,6 +143,28 @@ $ clang \
...
+Users should be warned that Omax.cfg enables `-ffast-math` which breaks IEEE compliance and
+enables maths optimisations which can affect code correctness.  LTOs are
+kept separately in OmaxLTO.cfg as users may not want LTOs due to potential increase in link time
+and/or increased memory usage during linking. The flags used in these configs are used internally by
+us at Arm, information on what they do can be found at
```
Comment (voltur01) [inline README.md:164]: Thank you for the update! Sorry, I was not clear: I meant that these are clang internal options (-mllvm ones) and I believe they are not documented - best I was able to find for some of them was comments in the source code. Maybe this is the way to put it: config files use llvm internal undocumented options, please see source code of corresponding optimization passes in LLVM project to find out more.

Comment (nasherm) [reply README.md]: Ah, I see. Updated

**Review by voltur01 — APPROVED** [review]
Looks good to me, thank you for the quick turnaround!

**Pull request merged**